src/nnbench/types.py

```diff
@@ -6,15 +6,31 @@
 import copy
 import inspect
 import os
+import shutil
+import weakref
 from abc import ABCMeta, abstractmethod
 from dataclasses import dataclass, field
-from typing import Any, Callable, Generic, Iterable, Iterator, Literal, TypeVar
+from pathlib import Path
+from tempfile import mkdtemp
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Generic,
+    Iterable,
+    Iterator,
+    Literal,
+    TypeVar,
+)
 
 from nnbench.context import Context
 
 T = TypeVar("T")
 Variable = tuple[str, type, Any]
 
+if TYPE_CHECKING:
+    import fsspec
```

Comment:
Is this mypy complaining? You could also add `fsspec` to the list of modules for which to ignore missing imports (in the pyproject.toml) if you like, since most of its interfaces are not great typing-wise.